stats/stats_test.go

```diff
@@ -724,3 +724,10 @@ func TestNaN(t *testing.T) {
 // should get 3 buckets 0-1 with count 1
 // 1-2 with count 3
 // 2-2.5 with count 1
+
+func BenchmarkBucketLookUpWithHighestValue(b *testing.B) {
+	testHistogram := NewHistogram(0, 1)
+	for i := 0; i < b.N; i++ {
+		testHistogram.Record(100000)
```

Comment:
add another one with a bit more varied data (maybe rand though that may test the speed of rand more than histogram’s speed)
and say what you get before/after

Reply:
btw the value that is slowest is lastValue-1 so 99999 not 100000

Reply:
Last value is selected from histogramBucketsValues array length - 1 which maps to 100000 as value. So the value that I used, allows me to iterate over all values. If scaledVal >= lastValue  is like this in record, you were right.